### DISCUSSION TITLE
How do I use conditionals in v3?

### DISCUSSION THREAD

**Discussion by turbostackapp** [muhammadsammy/tailwindcss-classnames#366] [Q&A] [answered]
I feel a little embarassed but I'm trying out the lib and couldn't figure out how to use conditionals in v3.

I tried the v2 way presented in the readme but it doesn't work.

This works fine:

```
export const container = (someFlag: boolean) =>
  classnames(
    width(collapsed ? 'w-20' : 'w-72')
  );
```

But what if I just want to add a class if `someFlag` is true? This doesn't work:

```
export const container = (someFlag: boolean) =>
  classnames(
    display('flex'),
    {
      [classnames(padding('p-5'))]: someFlag
    }
  );
```

I also tried calling `padding` without the nested `classnames` but it didn't work aswell. In both cases I get:

```
Argument of type '{ TAILWIND_STRING: any; }' is not assignable to parameter of type 'TArg'
```

How are you guys using it?

Thank you.

**Comment by bstro** [comment, marked as answer]
In my experience, this seems to work:
```
  classnames(
    display("flex"),
    padding({
      "p-5": someFlag
    })
  )
```
